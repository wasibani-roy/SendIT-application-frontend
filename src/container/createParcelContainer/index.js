import React, { Component } from 'react';
import Parcel from '../../components/createParcelComponent';
import CreateParcel from '../../actions/createParcelAction';

class ParcelContainer extends Component {
  state = {
    parcel_name: '',
    receiver: '',
    destination: '',
    price: '',
  };

  handleText = (event) => {
    this.setState({ [event.target.id]: event.target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    CreateParcel(this.state, this.props);
  };

  render() {
    return (
      <div className="login background-image content-wrapper">
        <Parcel
          handleText={this.handleText}
          handleSubmit={this.handleSubmit}
        />
      </div>
    );
  }
}

export default ParcelContainer;
