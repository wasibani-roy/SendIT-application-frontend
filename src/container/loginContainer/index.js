import React, { Component } from 'react';
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { toast } from "react-toastify";
import CircularProgressLoader from "../../commons/progressLoader";
import LoginAction from "../../actions/loginAction";
import Navbar from "../../components/navBarComponent"
import Login from "../../components/loginComponent";

const jwt = require("jsonwebtoken");

export class LoginContainer extends Component {
  state = {
    loader: {
      success: false,
      loading: false,
    },
    isProcessing: false,
  };

  componentWillReceiveProps(nextProps) {
    if (nextProps.token) {
      const decoded = jwt.decode(nextProps.token);
      console.log(decoded);
      if (decoded) {
        const userID = decoded.identity.user_id;
        localStorage.setItem("userID", userID);
        this.props.history.push("/parcels");
        toast.success("You have successfully signed in", {
          position: toast.POSITION.TOP_CENTER,
          autoClose: 3000,
          hideProgressBar: false,
          pauseOnHover: true,
        });
        setTimeout(() => window.location.reload('/create'), 100);
      }
    }
    this.setState({ loader: { loading: false } });
    this.setState({ isProcessing: false });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    const { LoginAction } = this.props;
    const userData = {
      email: event.target.elements.email.value,
      password: event.target.elements.password.value,
    };
  
    LoginAction(userData);
    this.setState({ loader: { loading: true } });
    this.setState({ isProcessing: true });
  };

  render() {
    const {errors} = this.props;
    const loader = this.state.loader;
    const buttonDisable = this.state.isProcessing;
    return (
        <div className="login">
            <Navbar />
            <CircularProgressLoader {...loader} />
            <Login
                handleChange={this.handleChange}
                errors={errors}
                isProcessing={buttonDisable}
                handleSubmit={this.handleSubmit}
            />
        </div>
    );
  }
}

export const mapStateToProps = state => ({
  token: state.loginReducer.token,
  errors: state.loginReducer.errors,
  isSuccessful: state.loginReducer.isSuccessful,
});

LoginContainer.propTypes = {
  token: PropTypes.string,
  errors: PropTypes.string
};

LoginContainer.defaultProps = {
  token: "",
  errors: "",
};

export default connect(mapStateToProps, { LoginAction})(LoginContainer);
