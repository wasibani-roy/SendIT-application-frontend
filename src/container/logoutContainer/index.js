import React, { Component } from "react";
import { toast, ToastContainer } from "react-toastify";

export class LogoutPage extends Component {
  componentWillMount() {
    localStorage.removeItem("access_token");
    localStorage.removeItem("userID");
    this.props.history.push("/");
    toast.success("You have successfully Logged out", {
      position: toast.POSITION.BOTTOM_RIGHT,
      autoClose: 3000,
      hideProgressBar: false,
      pauseOnHover: true,
    });
  }

  render() {
    return (
      <div>
        <ToastContainer />
      </div>
    );
  }
}

export default LogoutPage;
