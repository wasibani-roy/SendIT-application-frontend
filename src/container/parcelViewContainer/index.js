import React, { Component } from "react";
import { connect } from "react-redux";
import { ParcelItem } from "../../components/parcelItemComponent";
import fetchParcelsAction from "../../actions/fetchParcelAction";
import Navbar from "../../components/navBarComponent"

export class Parcels extends Component {
  componentWillMount = () => {
    this.props.fetchParcelsAction();
  };

  render() {
    const { parcels } = this.props;

    const parcelItems = parcels.map((parcel) => {
      return <ParcelItem key={parcel.parcel_name} {...parcel} />;
    });

    return (
      <React.Fragment>
      <Navbar />
      <div className="col-lg-8">
        <div className="article-container">{ parcelItems }</div>
      </div>
      </React.Fragment>
    );
  }
}

export const mapStateToProps = state => ({
    parcels: state.fetchParcelsReducer.parcels,
});

export default connect(
  mapStateToProps,
  { fetchParcelsAction },
)(Parcels);
