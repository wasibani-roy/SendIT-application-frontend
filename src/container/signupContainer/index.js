import React, { Component } from 'react';
import SignUp from '../../components/signupComponent';
import SignupAction from '../../actions/signupAction';
import Navbar from '../../components/navBarComponent';

class SiginupContainer extends Component {
  state = {
    username: '',
    email: '',
    password: '',
    role:'',
  };

  handleChange = (event) => {
    this.setState({ [event.target.id]: event.target.value });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    SignupAction(this.state, this.props);
  };

  render() {
    return (
        <div className="login">
            <Navbar />
            <SignUp
                handleChange={this.handleChange}
                handleSubmit={this.handleSubmit}
            />
        </div>
    );
  }
}

export default SiginupContainer;
