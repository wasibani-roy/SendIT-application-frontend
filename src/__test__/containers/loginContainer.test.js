import React from "react";
import { shallow } from "enzyme";
import { LoginContainer, mapStateToProps } from "../../container/loginContainer";

const props = {
  LoginAction: jest.fn(),
  history: {
    push: jest.fn(),
  },
};

const user = {
  identity:{
    user_id:17,
  },
  email: "roy@andela.com",
  password: "spidermansucks123",
};

const generateToken = (user) => {
  const jwt = require("jsonwebtoken");
  const token = jwt.sign(user, "sfhjjkdfjdkdfjdk", { expiresIn: 60 * 60 * 24 });
  return token; 
};

const token = generateToken(user);

describe("LoginView", () => {
  const wrapper = shallow(<LoginContainer {...props} />);
  it("should render without crashing", () => {
    const props = {};
    const instance = new LoginContainer(props);
    const wrapper = shallow(instance.render());
    expect(wrapper).toMatchSnapshot();
  });

  it("should redirect on new token", () => {
    const newProps = {
      ...props,
      token,
    };
    const wrapper = shallow(<LoginContainer {...props} />);
    const spyProp = jest.spyOn(wrapper.instance().props.history, "push");
    wrapper.setProps(newProps);
    expect(spyProp).toHaveBeenCalled();
  });
  
  it("should call the login action", () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        elements: {
          email: {
            value: "testemail@mail.com",
          },
          password: {
            value: "Password1",
          },
        },
      },
    };
    wrapper.find("Login").dive().find("form")
      .simulate("submit", event);
    const spyProp = jest.spyOn(wrapper.instance().props, "LoginAction");
    expect(spyProp).toHaveBeenCalled();
  });

  it("should map state to props", () => {
    const state = {
      loginReducer: {
        token: "",
        errors: {},
        isSuccessful: true,
      },
    };
    expect(mapStateToProps(state)).toEqual({
      token: "",
      errors: {},
      isSuccessful: true,
    });
  });
});
