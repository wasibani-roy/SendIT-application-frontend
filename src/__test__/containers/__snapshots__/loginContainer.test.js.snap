// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div
    className="login"
  >
    <Navbar />
    <CircularProgressLoader
      loading={false}
      success={false}
    />
    <Login
      errors=""
      handleSubmit={[Function]}
      isProcessing={false}
    />
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Navbar />,
        <CircularProgressLoader
          loading={false}
          success={false}
        />,
        <Login
          errors=""
          handleSubmit={[Function]}
          isProcessing={false}
        />,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "loading": false,
          "success": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "errors": "",
          "handleChange": undefined,
          "handleSubmit": [Function],
          "isProcessing": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Navbar />,
          <CircularProgressLoader
            loading={false}
            success={false}
          />,
          <Login
            errors=""
            handleSubmit={[Function]}
            isProcessing={false}
          />,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "loading": false,
            "success": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "errors": "",
            "handleChange": undefined,
            "handleSubmit": [Function],
            "isProcessing": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
